main/src/ca/uwaterloo/flix/language/phase/Weeder.scala

```diff
@@ -699,6 +699,10 @@ object Weeder {
         case e => WeededAst.Expression.Discard(e, loc)
       }
 
+    case ParsedAst.Expression.ForEach(sp1, pat, exp1, exp2, sp2) =>
+      val loc = mkSL(sp1, sp2)
```

Comment:
For simplicity, you can do an unsafe pattern match on `pat` to only handle the case where it is a variable. E.g. something like:

```
val varSym = pat match {
  case Pattern.Var(...) => ..
  case _ => ???
```

Then you recursively visit exp1 and exp2.

Then you need to construct a call `List.foreach(varSym -> exp1, exp2)` which means you need to construct some applies nodes and a Def node.